File: 392-is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True

        slow = 0
        for fast in range(len(t)):
            if t[fast] == s[slow]:
                slow += 1
        
            if slow == len(s):
                return True

        return False

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.equals("")) {
            return true;
        }
        if (t.equals("")) {
            return false;
        }

        int slow = 0;
        for (int fast = 0; fast < t.length(); fast++) {
            if (s.charAt(slow) == t.charAt(fast)) {
                slow++;
            }
            if (slow == s.length()) {
                return true;
            }
        }
        return false;
    }
}